app/uk/gov/hmrc/tai/forms/income/previousYears/UpdateIncomeDetailsForm.scala

```diff
@@ -47,8 +47,8 @@ object UpdateIncomeDetailsForm {
 
   def textExceedsCharacterLimit(exceedErrorMsg: String): Constraint[String] =
     Constraint[String]("characterLimitExceeded") {
-      case textValue if textValue.trim.length <= historicEmploymentDetailsCharLimit => Valid
-      case _                                                                        => Invalid(exceedErrorMsg)
+      case textValue if textValue.trim.replace("\r", "").length <= historicEmploymentDetailsCharLimit => Valid
```

Comment:
Missing a test, also I believe this is the same as doing `stripLineEnd` ?

Reply:
stripLineEnd doesn't work of this, it only strips a trailing new line at the end of the whole string, it wouldn't handle new lines in the middle of the text

Reply:
spec added